File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    const reporting = {};
    const reported = {};
    for(let i = 0; i < id_list.length; i++){
        const user = id_list[i];
        reporting[user] = [];
        reported[user] = 0;
    }
    for(let i = 0; i < report.length; i++){
        const [start, end] = report[i].split(' ');
        // start가 end를 신고한 기록이 없을 경우에만
        if(!reporting[start].includes(end)){
            reporting[start].push(end);
            reported[end]++;
        } 
    }
    const answer = [];
    const entries = Object.entries(reporting);
    for(const [start, ends] of entries){
        const number = ends.filter((user) => reported[user] >= k).length;
        answer.push(number)
    }
    return answer;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    reporting = {}
    reported = {}
    for id in id_list:
        reporting[id] = set()
        reported[id] = 0
    for cmd in report:
        o, d = cmd.split(' ')
        if not d in reporting[o]:
            reporting[o].add(d)
            reported[d] += 1
            
    answer = []
    for id in id_list:
        count = 0
        report_set = reporting[id]
        for d in report_set:
            if reported[d] >= k:
                count += 1
        answer.append(count)
    
    return answer